bundle/src/main/java/com/adobe/acs/commons/cloudservices/pwa/impl/ConfigurationImpl.java

```diff
@@ -1,36 +1,121 @@
+    private static final String RESOURCE_TYPE = "acs-commons/cloudservices/pwa/components/conf/page";
+    private static final String PN_SERVICE_WORKER_JS_CLIENT_LIBRARY_CATEGORY = "serviceWorkerJsClientLibraryCategory";
+    private static final String PN_PWA_JS_CLIENT_LIBRARY_CATEGORY = "pwaJsClientLibraryCategory";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+
+    @OSGiService
+    private SlingSettingsService slingSettingsService;
+
+    private Page currentPage;
+    private PageManager pageManager;
+
+    // These are resolved in init()
+    private Page rootPage;
+    private Page confPage;
+
+    @PostConstruct
+    protected void init() {
+        pageManager = resourceResolver.adaptTo(PageManager.class);
+        currentPage = pageManager.getContainingPage(request.getResource());
 
+        Page page = currentPage;
 
-    public boolean show() {
-        Resource jcrContent = configResource.getChild(JcrConstants.JCR_CONTENT);
-        if (jcrContent != null) {
-            if (jcrContent.isResourceType(RESOURCE_TYPE)) {
-                return true;
+        while (page != null) {
+            final String confPath = page.getProperties().get("cq:conf", String.class);
+
+            if (StringUtils.isNotBlank(confPath)) {
+                Resource cloudConfigsResource = resourceResolver.getResource(confPath + "/settings/cloudconfigs");
+                if (cloudConfigsResource != null) {
+                    for (final Resource cloudConfigResource : cloudConfigsResource.getChildren()) {
+                        Resource jcrContentResource = cloudConfigResource.getChild(JcrConstants.JCR_CONTENT);
+                        if (jcrContentResource != null && jcrContentResource.isResourceType(RESOURCE_TYPE)) {
+                            confPage = pageManager.getContainingPage(jcrContentResource);
+                            rootPage = page;
+                            return;
+                        }
+                    }
+                }
             }
+
+            page = page.getParent();
         }
+    }
 
-        return super.show();
+    @Override
+    public boolean isReady() {
+        return true;
+        // TODO Uncomment
+        //return slingSettingsService.getRunModes().contains("publish");
     }
 
+    @Override
+    public ValueMap getProperties() {
+        return confPage.getContentResource().getValueMap();
+    }
 
     @Override
-    public String getIcon() {
-        return "gauge5";
+    @Nullable
+    public String getScopePath() {
+        if (rootPage != null) {
+            return resourceResolver.map(request, rootPage.getPath());
+        }
+
+        return null;
+    }
+
+    @Override
+    @Nullable
+    public String[] getServiceWorkerJsCategories() {
+        if (confPage != null) {
+            return confPage.getProperties().get(PN_SERVICE_WORKER_JS_CLIENT_LIBRARY_CATEGORY, String[].class);
```

Comment:
@davidjgonzalez This & PN_PWA_JS_CLIENT_LIBRARY_CATEGORY is a single valued in the wizard pwa wizard

Reply:
Cool - think i forgot if it was 1 or possibly many, since a CLib can have multiple categories.